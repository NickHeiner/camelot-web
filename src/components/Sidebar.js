import React, {PureComponent} from 'react';
import autobind from 'autobind-decorator';
import { Button } from 'react-bootstrap';
import firebase from 'firebase';
import {browserHistory} from 'react-router';
import { slide as SlideMenu } from 'react-burger-menu';
import Avatar from './Avatar';

class Sidebar extends PureComponent {
    static propTypes = {
        currentUser: React.PropTypes.object
    }

    constructor() {
        super();
        this.auth = new firebase.auth();
    }

    render() {
        const currentUser = this.props.currentUser;
        return (
            <SlideMenu>
                <div className="user-bar">
                    <span>{currentUser.displayName}</span>
                    <Avatar currentUser={currentUser} />
                </div>
                <Button bsStyle="primary" onClick={this.signOut}>Sign Out</Button>
            </SlideMenu>
        );
    }

    @autobind
    signOut() {
        this.auth.signOut().then(() => browserHistory.push('/'));
    }
}

export default Sidebar;



