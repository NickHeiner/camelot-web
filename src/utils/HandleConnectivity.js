import {connect} from 'react-redux';
import React, {PureComponent} from 'react';
import _ from 'lodash';

export default (OnlineContainer, OfflineContainer) => {
    @connect(
        state => _.pick(state, 'offline')
    )
    class HandleConnectivity extends PureComponent {
        render = () => this.props.offline ? <OfflineContainer /> : <OnlineContainer />;
    };
    
    return HandleConnectivity;
};
