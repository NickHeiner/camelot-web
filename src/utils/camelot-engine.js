import {Iterable} from 'immutable';
// This is a bit hacky but w/e.
import pairwise from 'camelot-engine/lib/util/pairwise';
import camelotEngine from 'camelot-engine';
const query = camelotEngine().query();

const ensureJS = val => Iterable.isIterable(val) ? val.toJS() : val;

// TODO clean this up by always using rest params
const proxyWithJsArgs = fn => (...args) => fn(...args.map(ensureJS));

export const isValidMove = (gameState, possibleMoveSteps, movingPlayerName) =>
  query.isValidMove(ensureJS(gameState), ensureJS(possibleMoveSteps), movingPlayerName);

export const getBoardSpace = (...args) =>
  query.getBoardSpace(...args.map(ensureJS));

export const isGoal = (gameState, ...rest) =>
  query.isGoal(ensureJS(gameState), ...rest);

export const getPairs = proxyWithJsArgs(pairwise);
export const getCoordsBetween = proxyWithJsArgs(query.getCoordsBetween);
