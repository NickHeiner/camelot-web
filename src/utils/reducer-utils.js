
export const getCurrentGame = (fullState, gameId) => fullState.firebase.getIn(['data', 'games', gameId]);

export const getCurrentUserPlayerName = (fullState, gameId) => {
  const currentUserUid = fullState.firebase.getIn(['profile', 'uid']);
  const currentGame = getCurrentGame(fullState, gameId);
  const currentHostUid = fullState.firebase.getIn(['data', 'users', currentGame.get('host'), 'uid']);
  const currentOpponentUid = fullState.firebase.getIn(['data', 'users', currentGame.get('opponent'), 'uid']);

  if (currentUserUid === currentHostUid) {
    return 'playerA';
  }
  if (currentUserUid === currentOpponentUid) {
    return 'playerB';
  }
  return null;
};
