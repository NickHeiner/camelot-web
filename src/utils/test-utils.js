export const getDummyCurrentUser = () => ({
    photoURL: 'photoURL',
    displayName: 'Display Name'
});
