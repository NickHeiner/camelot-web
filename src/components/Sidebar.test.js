import {SidebarPresentation} from './Sidebar';
import React from 'React';
import { shallow } from 'enzyme';
import { getDummyCurrentUser } from '../utils/test-utils';

describe('Sidebar', () => {
    it('renders', () => {
        const currentUser = getDummyCurrentUser();
        expect(shallow(
            <SidebarPresentation currentUser={currentUser} />
        ).html()).toMatchSnapshot();
    });
});
