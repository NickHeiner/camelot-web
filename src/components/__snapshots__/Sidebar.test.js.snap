exports[`Sidebar renders 1`] = `"<div data-radium=\"true\"><div class=\"bm-overlay\" style=\"position:fixed;z-index:1;width:100%;height:100%;background:rgba(0, 0, 0, 0.3);opacity:0;transform:translate3d(100%, 0, 0);transition:opacity 0.3s, transform 0s 0.3s;\" data-radium=\"true\"></div><div id=\"\" class=\"bm-menu-wrap\" style=\"position:fixed;right:inherit;z-index:2;width:300px;height:100%;transform:translate3d(-100%, 0, 0);transition:all 0.5s;\" data-radium=\"true\"><div class=\"bm-menu\" style=\"height:100%;box-sizing:border-box;overflow:auto;\" data-radium=\"true\"><nav class=\"bm-item-list\" style=\"height:100%;\" data-radium=\"true\"><div class=\"user-bar\" style=\"display:block;outline:none;\" data-radium=\"true\"><span>Display Name</span><img src=\"photoURL\" alt=\"avatar for Display Name\" class=\"profile-picture\"/></div><button style=\"0:[object Object];\" type=\"button\" class=\"btn btn-primary\">Sign Out</button></nav></div><div data-radium=\"true\"><div class=\"bm-cross-button\" style=\"position:absolute;width:24px;height:24px;right:8px;top:8px;\" data-radium=\"true\"><span style=\"position:absolute;top:6px;right:14px;\" data-radium=\"true\"><span class=\"bm-cross\" style=\"position:absolute;width:3px;height:14px;transform:rotate(45deg);\" data-radium=\"true\"></span><span class=\"bm-cross\" style=\"position:absolute;width:3px;height:14px;transform:rotate(-45deg);\" data-radium=\"true\"></span></span><button style=\"position:absolute;left:0;top:0;width:100%;height:100%;margin:0;padding:0;border:none;text-indent:-9999px;background:transparent;outline:none;\" data-radium=\"true\">Close Menu</button></div></div></div><div class=\"bm-burger-button\" style=\"z-index:1;\" data-radium=\"true\"><span data-radium=\"true\"><span class=\"bm-burger-bars\" style=\"position:absolute;height:20%;left:0;right:0;top:0%;opacity:1;\" data-radium=\"true\"></span><span class=\"bm-burger-bars\" style=\"position:absolute;height:20%;left:0;right:0;top:40%;opacity:1;\" data-radium=\"true\"></span><span class=\"bm-burger-bars\" style=\"position:absolute;height:20%;left:0;right:0;top:80%;opacity:1;\" data-radium=\"true\"></span></span><button style=\"position:absolute;left:0;top:0;width:100%;height:100%;margin:0;padding:0;border:none;opacity:0;font-size:8px;\" data-radium=\"true\">Open Menu</button></div></div>"`;
